code/modules/reagents/reagents/vg.dm

```diff
@@ -13,24 +13,29 @@ var/global/list/charcoal_doesnt_remove=list(
 	reagent_state = LIQUID
 	color = "#333333" // rgb: 51, 51, 51
 	custom_metabolism = 0.06
-	digestion_rate = 0 // will not transfer from stomach to body
-
-/datum/reagent/charcoal/digest(var/mob/living/carbon/human/M)
-	var/datum/organ/internal/stomach/S = M.get_stomach()
```

Comment:
And this is the other one
This would fix the bug where charcoal just does nothing for people without a stomach